src/compiler/c.rs

```diff
@@ -323,7 +325,27 @@ pub fn hash_key(compiler_digest: &str,
             val.hash(&mut HashToDigest { digest: &mut m });
         }
     }
-    m.update(preprocessor_output);
+
+    if let Ok(strip_dirs) = env::var("SCCACHE_STRIP_DIRS") {
+        trace!("Attempting to strip dirs from preprocessor output");
+        m.update(strip_dirs.as_bytes());
```

Comment:
Now that you say that, yeah, it looks like it wouldn't work, but for some reason it worked locally with my testing 🤔 Probably because my strip_dirs was `path1:path2` 🤔 Will remove this from hash generation